File: PHP/project/src/CH8/Sec2_2/LogAnalyzer.php
<?php

namespace Src\CH8\Sec2_2;


use Exception;

class LogAnalyzer
{
    /**
     * @var bool $initialize 是否初始化
     */
    private $initialize = false;

    public function initialize(): void
    {
        // do something initialize
        $this->initialize = true;
    }

    public function isValid(string $fileName): bool
    {
        if ( ! $this->initialize) {
            throw new Exception('not initialize');
        }

        if (strlen($fileName) < 8) {
            return true;
        }
        return false;
    }
}

File: C#/CKUT/CKUT.Core/Services/Ch8/8.2.2/LogAnalyzer.cs
namespace CKUT.Core.Services.Ch8._8._2._2
{
    public class LogAnalyzer
    {
        public bool IsValid(string fileName)
        {
            if (fileName.Length < 8)
            {
                return true;
            }

            return false;
        }
    }
}